.travis.yml

```diff
@@ -0,0 +1,37 @@
+language: scala
+scala:
+   - 2.11.8
+   - 2.12.0-M4
+
+jdk:
+  - oraclejdk8
+
+matrix:
+  allow_failures:
+    - scala: 2.12.0-M4
+    - jdk: openjdk8
+
+sudo: false
+
+before_cache:
+  # Tricks to avoid unnecessary cache updates
+  - find $HOME/.ivy2 -name "ivydata-*.properties" -delete
+  - find $HOME/.sbt -name "*.lock" -delete
+
+cache:
+  directories:
+    - $HOME/.ivy2/cache
+    - $HOME/.sbt/boot/
+    - ui/node_modules
+env:
+  - TRAVIS_NODE_VERSION="4"
+  - TRAVIS_NODE_VERSION="5"
+
+install:
+  - rm -rf ~/.nvm && git clone https://github.com/creationix/nvm.git ~/.nvm && (cd ~/.nvm && git checkout `git describe --abbrev=0 --tags`) && source ~/.nvm/nvm.sh && nvm install $TRAVIS_NODE_VERSION
+  - npm set progress=false -g > /dev/null 2>&1
+  - (cd ui;npm install)
+    
+script:
+   - sbt ++$TRAVIS_SCALA_VERSION -Dfile.encoding=UTF8 -J-XX:ReservedCodeCacheSize=256M test
```

Comment:
Because that's what is [recommended by SBT guys](http://www.scala-sbt.org/0.13/docs/Travis-CI-with-sbt.html)

AFAIK the thing is to reduce memory needed by the JVM and guarantee the same setting between JVM versions